### PR TITLE
WFLY-19664 Log warning if hibernate.cache.region.factory_class setting is ignored

### PR DESCRIPTION
**Pull request by scottmarlow**
Wildfly issue links:
> * [WFLY-19664](https://issues.redhat.com/browse/WFLY-19664)

### CODE REVIEW

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @scottmarlow

**Comment by scottmarlow** [comment]
fyi @Sanne @yrodiere with this change, WildFly 34 will log a warning if the hibernate.cache.region.factory_class setting is ignored due to the JPA/Persistence container causing it to be ignored.

**Comment by Sanne** [comment]
thanks @scottmarlow !

File: jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/JpaLogger.java [hunk 1]
```diff
@@ -46,6 +47,10 @@ public interface JpaLogger extends BasicLogger {
     @Message(id = 20261, value = "Hibernate ORM did not register LifeCycleListener")
     IllegalStateException HibernateORMDidNotRegisterLifeCycleListener();
 
+    @LogMessage(level = WARN)
+    @Message(id = 20262, value = "Application custom cache region setting is ignored %s=%s")
+    void ignoredCacheRegionSetting(String propertyName, String setting );
```
Comment (Sanne) [inline jpa/hibernate6/src/main/java/org/jboss/as/jpa/hibernate/JpaLogger.java:52]: maybe the message could suggest which property is expected to be used instead in Wildfly?

**Pull request merged**